prov/util/src/util_mr_cache.c

```diff
@@ -149,6 +181,7 @@ bool ofi_mr_cache_flush(struct ofi_mr_cache *cache)
 	bool empty;
 
 	pthread_mutex_lock(&cache->monitor->lock);
+	util_mr_process_stale_entries(cache);
```

Comment:
We can't hold the monitor lock during any entry cleanup.  That can still result in a hang.

Reply:
I've moved the code outside of the lock again.